src/main/java/com/thebois/models/IStructureFinder.java

```diff
@@ -0,0 +1,33 @@
+package com.thebois.models;
+
+import java.util.Optional;
+
+import com.thebois.models.world.structures.IStructure;
+import com.thebois.models.world.structures.StructureType;
+
+/**
+ * Allows for locating of specific types of objects.
+ */
+public interface IStructureFinder {
+
+    /**
+     * Finds and returns the closest IStructure to the passed position.
+     *
+     * @param position The position to find a structure closest to.
+     * @param type The type of structure to find.
+     *
+     * @return The found IStructure.
+     */
+    Optional<IStructure> findNearestStructure(Position position, StructureType type);
+
+    /**
+     * Finds and returns the closest incomplete IStructure to the passed position.
+     *
+     * @param position The position to find an incomplete structure closest to.
+     * @param type The type of structure to find.
+     *
+     * @return The found IStructure.
+     */
+    Optional<IStructure> findNearestIncompleteStructure(Position position, StructureType type);
```

Comment:
Sorry if I'm annoying, but this method should not have a StructureType since builders don't care what structure to build, just that it's not completed.

Reply:
I could see either way being useful, what if a specific role is tasked with only building a certain types of building? But then again, we would need a type for "any" building...

Reply:
Don't think you should have a type for all/any type. Or at least hope not, since that would break the structures button in my pr, since the buttons are based on the enum. 
Maybe have both methods if you can see a need for both of them.

Reply:
I removed the StructureType from the method. If we need it, we can add it again later.